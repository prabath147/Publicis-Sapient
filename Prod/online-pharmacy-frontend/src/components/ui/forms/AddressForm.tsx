import { Space, Text, TextInput } from "@mantine/core";
import { UseFormReturnType } from "@mantine/form";

interface addressFormPropType {
    form: UseFormReturnType<any>,
    formFieldName?: string
}
export default function AddressForm({ form, formFieldName = "address" }: addressFormPropType) {
    return (
        <div>
            <Space />
            <Text >Address</Text>
            <TextInput label="Street" placeholder='Street' withAsterisk {...form.getInputProps(formFieldName + '.street')} />
            <TextInput label="City" placeholder='City' withAsterisk {...form.getInputProps(formFieldName + '.city')} />
            <TextInput label="State" placeholder='State' withAsterisk {...form.getInputProps(formFieldName + '.state')} />
            <TextInput label="Country" placeholder='Country' withAsterisk {...form.getInputProps(formFieldName + '.country')} />
            <TextInput type="number" label="Pincode" placeholder="PinCode" withAsterisk {...form.getInputProps(formFieldName + '.pinCode')} />
        </div>
    )
}

export const addressValidation = (): any => {
    return {
        city: value => value.length < 2 ? "This field is required!" : null,
        country: value => value.length < 2 ? "This field is required!" : null,
        pinCode: value => Math.floor(value) < 99999 ? "Valid Pincode is required!" : null,
        state: value => value.length < 2 ? "This field is required!" : null,
        street: value => value.length < 2 ? "This field is required!" : null,
    }
}

export const addressInitialValues = (): AddressInterface => {
    return {
        city: '',
        country: '',
        pinCode: '',
        state: '',
        street: ''
    }
}

export interface AddressInterface {
    addressId?: number;
    city: string,
    country: string,
    pinCode: string,
    state: string,
    street: string,
}

