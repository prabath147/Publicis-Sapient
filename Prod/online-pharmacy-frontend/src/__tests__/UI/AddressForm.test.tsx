import { useForm } from '@mantine/form';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AddressForm, { addressInitialValues, AddressInterface, addressValidation } from '../../components/ui/forms/AddressForm';

const MockAddressForm = () => {
    const form = useForm<AddressInterface>({
        initialValues: addressInitialValues(),
        validate: addressValidation()
    })
    return (
        <div>
            <AddressForm form={form} />
            <button onClick={() => {
                form.isValid()
            }}>place order</button>
        </div>
    )
}

describe('AddressForm unit tests', () => {

    it('component created', () => {
        const component = renderer.create(<MockAddressForm />)
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })


    it('test validations', async () => {
        const { asFragment } = render(<MockAddressForm />)

        expect(screen.getByRole('textbox', { name: /Street/i })).toBeInTheDocument();
        fireEvent.click(screen.getByText(/place order/i))

        expect(asFragment()).toMatchSnapshot();
    })

})