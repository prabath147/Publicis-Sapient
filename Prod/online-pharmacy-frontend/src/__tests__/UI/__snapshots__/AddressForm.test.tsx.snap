// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm unit tests component created 1`] = `
<div>
  <div>
    <div
      className="mantine-a9v14h"
    />
    <div
      className="mantine-Text-root mantine-1pbxw0k"
    >
      Address
    </div>
    <div
      className="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <label
        className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
        htmlFor="mantine-r0"
        id="mantine-r0-label"
      >
        Street
        <span
          aria-hidden={true}
          className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid={false}
          className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
          id="mantine-r0"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Street"
          type="text"
        />
      </div>
    </div>
    <div
      className="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <label
        className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
        htmlFor="mantine-r1"
        id="mantine-r1-label"
      >
        City
        <span
          aria-hidden={true}
          className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid={false}
          className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
          id="mantine-r1"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="City"
          type="text"
        />
      </div>
    </div>
    <div
      className="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <label
        className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
        htmlFor="mantine-r2"
        id="mantine-r2-label"
      >
        State
        <span
          aria-hidden={true}
          className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid={false}
          className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
          id="mantine-r2"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="State"
          type="text"
        />
      </div>
    </div>
    <div
      className="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <label
        className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
        htmlFor="mantine-r3"
        id="mantine-r3-label"
      >
        Country
        <span
          aria-hidden={true}
          className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid={false}
          className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
          id="mantine-r3"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Country"
          type="text"
        />
      </div>
    </div>
    <div
      className="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
    >
      <label
        className="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
        htmlFor="mantine-r4"
        id="mantine-r4-label"
      >
        Pincode
        <span
          aria-hidden={true}
          className="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
        >
           *
        </span>
      </label>
      <div
        className="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
      >
        <input
          aria-invalid={false}
          className="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
          id="mantine-r4"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="PinCode"
          type="number"
        />
      </div>
    </div>
  </div>
  <button
    onClick={[Function]}
  >
    place order
  </button>
</div>
`;

exports[`AddressForm unit tests test validations 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="mantine-a9v14h"
      />
      <div
        class="mantine-Text-root mantine-1pbxw0k"
      >
        Address
      </div>
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
          for="mantine-r0"
          id="mantine-r0-label"
        >
          Street
          <span
            aria-hidden="true"
            class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
            id="mantine-r0"
            placeholder="Street"
            type="text"
          />
        </div>
      </div>
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
          for="mantine-r1"
          id="mantine-r1-label"
        >
          City
          <span
            aria-hidden="true"
            class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
            id="mantine-r1"
            placeholder="City"
            type="text"
          />
        </div>
      </div>
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
          for="mantine-r2"
          id="mantine-r2-label"
        >
          State
          <span
            aria-hidden="true"
            class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
            id="mantine-r2"
            placeholder="State"
            type="text"
          />
        </div>
      </div>
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
          for="mantine-r3"
          id="mantine-r3-label"
        >
          Country
          <span
            aria-hidden="true"
            class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
            id="mantine-r3"
            placeholder="Country"
            type="text"
          />
        </div>
      </div>
      <div
        class="mantine-InputWrapper-root mantine-TextInput-root mantine-1m3pqry"
      >
        <label
          class="mantine-InputWrapper-label mantine-TextInput-label mantine-ittua2"
          for="mantine-r4"
          id="mantine-r4-label"
        >
          Pincode
          <span
            aria-hidden="true"
            class="mantine-u5apz8 mantine-InputWrapper-required mantine-TextInput-required"
          >
             *
          </span>
        </label>
        <div
          class="mantine-Input-wrapper mantine-TextInput-wrapper mantine-12sbrde"
        >
          <input
            aria-invalid="false"
            class="mantine-Input-input mantine-TextInput-input mantine-gqmpge"
            id="mantine-r4"
            placeholder="PinCode"
            type="number"
          />
        </div>
      </div>
    </div>
    <button>
      place order
    </button>
  </div>
</DocumentFragment>
`;
